src/components/DatePicker/index.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import moment from 'moment';
+import ExpensiTextInput from '../ExpensiTextInput';
+import CONST from '../../CONST';
+import {propTypes, defaultProps} from './datepickerPropTypes';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+
+const datePickerPropTypes = {
+    ...propTypes,
+    ...windowDimensionsPropTypes,
+};
+
+class Datepicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.raiseDateChange = this.raiseDateChange.bind(this);
+        this.showDatepicker = this.showDatepicker.bind(this);
+
+        /* We're using uncontrolled input otherwise it wont be possible to
+        * raise change events with a date value - each change will produce a date
+        * and make us reset the text input */
+        this.defaultValue = props.value
+            ? moment(props.value).format(CONST.DATE.MOMENT_FORMAT_STRING)
+            : '';
+    }
+
+    componentDidMount() {
+        if (this.inputRef) {
+            // Adds nice native datepicker on web/desktop. Not possible to set this through props
+            this.inputRef.setAttribute('type', 'date');
+        }
+    }
+
+    /**
+     * Trigger the `onChange` handler when the user input has a complete date or is cleared
+     * @param {String} text
+     */
+    raiseDateChange(text) {
+        if (!text) {
+            this.props.onChange(null);
+            return;
+        }
+
+        const asMoment = moment(text);
+        if (asMoment.isValid()) {
+            const asDate = asMoment.toDate();
+            this.props.onChange(asDate);
+        }
+    }
+
+    /**
+     * Pops the datepicker up when we focus this field. This only works on mWeb
+     * On mWeb the user needs to tap on the field again in order to bring the datepicker. But our current styles
+     * don't make this very obvious. To avoid confusion we open the datepicker when the user focuses the field
+     */
+    showDatepicker() {
+        if (this.inputRef) {
+            this.inputRef.click();
+        }
+    }
```

Comment:
This was mainly added due to very bad UX on mWeb Android (Chrome).
Tapping on the field would only highlight it. The datepicker would only appear if you tap on the input element but due to our styles it's wrapped by a `<div>` and it's actually smaller...

#### Before
https://user-images.githubusercontent.com/12156624/135150158-0337af85-645b-4bba-8de9-96fe56b7461f.mp4
- it might not be obvious but the picker would appear only when we tap below the field label near the middle of the field. And only after we've focused the field...

#### After
https://user-images.githubusercontent.com/12156624/135150307-ac5eed39-f650-4083-b793-171836008a23.mp4
- The picker appears as soon as we tap the field (similar to how it works on Android native) 
Tested the rest of the platforms and there aren't any issues